### DISCUSSION TITLE
Design system

### DISCUSSION THREAD

**Discussion by asherccohen** [tayv/Project-Bubblegum#42] [Ideas]
A companion to the Component Library is the Design System.

Helps centralising styles, makie better decisions for layouts, margins  typography and colours.

Suggestion:

- use recommended tools
- set up the definitions
- create a theme
- consume it in components

**Comment by asherccohen** [comment]
Here all it means is really:

- restructure folders to be more discoverable
- choose a tool out of the many and build a config that is readable across all components (eg colors tokens are consumed via the theme instead of passing values around)

**Comment by tayv** [comment]
I was planning to use the[ theme section](https://tailwindcss.com/docs/theme) in `tailwind.config.js` to define key styles that would be re-used. So something like:

````
module.exports = {
  theme: {
    colors: {
      primary: '#F87171',
      secondary: '#34D399',
      ...
    },
    fontFamily: {
      heading: ['Gotham', 'sans-serif'],
      body: ['Inter', 'sans-serif'],
      ...
    },
  },
};
````

Is this in-line with what you're thinking?

**Reply by asherccohen** [reply]
Yes, absolutely, it's a good start!

**Reply by asherccohen** [reply]
You can always keep this in a folder named "theme", at the same level as "components"

**Comment by tayv** [comment]
Also a rough idea of how I've been thinking of refactoring components folder. Not sure how you feel about flatter vs more nested structures.

````
|- Components
   |- forms
       |- atoms
           |- RadioButton.tsx
                ...
       |- molecules // Add this to keep RHF Controller fully separate from form components
           |- RadioButtonGroup.tsx
                ...
       |- wrappersRHF
           |- ControlledRadioButtonGroup.tsx // implement RHF Controller at this level
   |- ui
      |- layout // anything that impacts layout/spacing/etc.
         |- SectionContainer.tsx
         ...
      |- messages // anything that displays messages to the user
         |- TipMessage.tsx
        ...
 ````

**Reply by asherccohen** [reply]
Yes, that's one way, or to be more generic you can do something like:

`
|- Design System
       |- atoms //dumb components 
           |- Button.tsx
           |- RadioButton.tsx
                ...
       |- molecules //dumb but composed components 
           |- Card.tsx
           |- RadioButtonGroup.tsx
                ...
       |- organisms //more complex 
         |- forms //anything related to forms
           |- FormCheckbox.tsx
           |- FormRadioButtonGroup.tsx
         |- tables
            ...
        |- templates
          |- layout
            |- FullPage.tsx
            |- Header.ts
            |- Footer.tsx
            ...
          |- screen
            |- Dashboard.tsx
            |- Settings.tsx
            ....any other reusable full screen
            
        
` 

And by leveraging composition you'll have the baseline for anything you want to build!

**Reply by tayv** [reply]
Applied most of your suggestions in this pull request: https://github.com/tayv/Project-Bubblegum/pull/44

**Reply by tayv** [reply]
Followup: Second PR to add Jest: https://github.com/tayv/Project-Bubblegum/pull/45
